VstsTasks/AzureDtlCreateVM/task-funcs.ps1

```diff
@@ -417,15 +431,35 @@ function Wait-ApplyArtifacts
                 throw "Waited for more than $(ConvertTo-MinutesString $totalWaitMinutes). Failing the task."
             }
 
-            $continueWaiting = $true
-
             $vm = Get-AzureRmResource -ResourceId $ResourceId
-            if ($vm)
+            if (-not $vm)
             {
-                $provisioningState = $vm.Properties.provisioningState
-                $continueWaiting = $provisioningState -and ($provisioningState -eq 'ApplyingArtifacts' -or $provisioningState -eq 'UpgradingVmAgent')
+                throw "Unable to find VM with resource ID '$ResourceId'."
             }
 
+            $vmName = $vm.Name
+            if ($vm.ResourceName)
+            {
+                $vmLabName = $vm.ResourceName.Split('/')[0]
+                $vmFullName = $vm.ResourceName
+            }
+            else
+            {
+                $vmLabName = $(if ($vm.ParentResource){ $vm.ParentResource.Split('/')[-1] } else { $null })
+                $vmFullName = $(if ($vmLabName){ "$vmLabName/$vmName" } else { $vmName })
+            }
+            $vmResourceGroupName = $vm.ResourceGroupName
+            $vmResourceType = $vm.ResourceType
+
+            $vmDetails = Get-AzureRmResource -ApiVersion '2017-04-26-preview' -Name $vmFullName -ResourceGroupName $vmResourceGroupName -ResourceType $vmResourceType -ODataQuery '$expand=Properties($expand=Artifacts)'

```

Comment:
Can the expand to include artifacts not be used with the Get-AzureRmResource used with a resource Id?

Reply:
No need to get the basic information for the resource. There are no expansions happening in this call.